plugins/slo/pkg/slo/system.go

```diff
@@ -29,6 +29,10 @@ func (p *Plugin) UseAPIExtensions(intf system.ExtensionClientInterface) {
 	cc, err := intf.GetClientConn(p.ctx, "CortexAdmin", "AlertEndpoints")
 	if err != nil {
 		p.logger.Error("failed to get cortex admin client", "error", err)
+		if p.ctx.Err() != nil {
+			// Plugin is shutting down, don't exit
+			return
```

Comment:
We should probably standardize checking ctx.Err() everywhere we os.Exit(1) in plugin setup : example plugin, metrics plugin, logging plugin

Reply:
I agree, we should come up with a better way to handle fatal errors from plugins.

Reply:
https://github.com/rancher/opni/issues/1567